cpp/src/arrow/ipc/metadata_internal.h

```diff
@@ -198,7 +198,7 @@ Status WriteDictionaryMessage(
     const int64_t id, const int64_t length, const int64_t body_length,
     const std::shared_ptr<const KeyValueMetadata>& custom_metadata,
     const std::vector<FieldMetadata>& nodes, const std::vector<BufferMetadata>& buffers,
-    const IpcWriteOptions& options, std::shared_ptr<Buffer>* out);
+    const IpcWriteOptions& options, std::shared_ptr<Buffer>* out, bool isDelta);
```

Comment:
We should keep `std::shared<Buffer>* out` at the end of the arguments.

Reply:
Sounds reasonable. I have revised the code to place it after the id argument. (I have also revised other APIs accordingly, to make it consistent)